internal/throttler/threshold/controller.go

```diff
@@ -0,0 +1,54 @@
+package threshold
+
+import (
+	"context"
+
+	"github.com/openfga/openfga/internal/throttler"
+)
+
+type dispatchThrottlingThresholdType uint32
+
+const (
+	dispatchThrottlingThreshold dispatchThrottlingThresholdType = iota
+)
+
+type Config struct {
+	Enabled      bool
+	Throttler    throttler.Throttler
+	Threshold    uint32
+	MaxThreshold uint32
+}
+
+func ShouldThrottle(ctx context.Context, currentCount uint32, defaultThreshold uint32, maxThreshold uint32) bool {
+	threshold := defaultThreshold
+
+	if maxThreshold == 0 {
+		maxThreshold = defaultThreshold
+	}
+
+	thresholdInCtx := DispatchThrottlingThresholdFromContext(ctx)
+
+	if thresholdInCtx > 0 {
+		threshold = min(thresholdInCtx, maxThreshold)
+	}
+
+	return currentCount > threshold
+}
+
+// ContextWithDispatchThrottlingThreshold will save the dispatch throttling threshold in context.
+func ContextWithDispatchThrottlingThreshold(ctx context.Context, threshold uint32) context.Context {
```

Comment:
```suggestion
func ContextWithThrottlingThreshold(ctx context.Context, threshold uint32) context.Context {
```
This package isn't specific to dispatch.